### ISSUE TITLE
[CI] Plugin lib tests failing on windows

### ISSUE THREAD

**Issue by thecoop** [elastic/elasticsearch#92866]
ClassReadersTests.classMethod failing
NamedComponentScannerTests.classMethod failing

Both on various windows builds. Most likely similar to other classloader test failures with removing files

**Build scan:**
https://gradle-enterprise.elastic.co/s/77y6wven7vqry/tests/:libs:elasticsearch-plugin-scanner:test/org.elasticsearch.plugin.scanner.ClassReadersTests

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.plugin.scanner.ClassReadersTests&tests.test=classMethod

**Failure excerpt:**
```
java.io.IOException: Could not remove the following files (in the order of attempts):
   C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001\tempDir-005: java.nio.file.DirectoryNotEmptyException: C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001\tempDir-005
   C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001\tempDir-005\jars-dir: java.nio.file.AccessDeniedException: C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001\tempDir-005\jars-dir
   C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001: java.nio.file.DirectoryNotEmptyException: C:\Users\jenkins\workspace\platform-support\593\libs\plugin-scanner\build\testrun\test\temp\org.elasticsearch.plugin.scanner.ClassReadersTests_40C8C1BA1BA2E7E4-001

  at __randomizedtesting.SeedInfo.seed([40C8C1BA1BA2E7E4]:0)
  at org.apache.lucene.util.IOUtils.rm(IOUtils.java:341)
  at org.apache.lucene.tests.util.TestRuleTemporaryFilesCleanup.afterAlways(TestRuleTemporaryFilesCleanup.java:209)
  at com.carrotsearch.randomizedtesting.rules.TestRuleAdapter$1.afterAlways(TestRuleAdapter.java:31)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1589)

```

**Comment by mark-vieira** [comment]
This sounds very much like we're leaving open file handles. Windows doesn't like that.

**Comment by rjernst** [comment]
I think this PR will fix the issue (though I have not confirmed on windows yet)
https://github.com/elastic/elasticsearch/pull/92890

**Comment by davidkyle** [comment]
It's back!

https://gradle-enterprise.elastic.co/s/vlkabtviblryo

This build was on commit f21881369c849011e6aa2621d64627e4e0669d69 which contains the fix in #92890

https://gradle-enterprise.elastic.co/s/vlkabtviblryo/tests/:libs:elasticsearch-plugin-scanner:test/org.elasticsearch.plugin.scanner.NamedComponentScannerTests
https://gradle-enterprise.elastic.co/s/vlkabtviblryo/tests/:libs:elasticsearch-plugin-scanner:test/org.elasticsearch.plugin.scanner.ClassReadersTests

**Comment by thecoop** [comment]
Think that error is #92782, not this issue
